src/DataSource/ArrayDataSource.php

```diff
@@ -220,7 +220,10 @@ public function applyFilterDateRange($row, FilterDateRange $filter)
 
 		if ($values['from'] !== NULL && $values['from'] !== '') {
 			$date_from = \DateTime::createFromFormat($format, $values['from']);
-
+			if (!($date_from instanceof \DateTime)) {
```

Comment:
Few lines below is a try-catch block, that tries to create `\DateTime` object from other formats. What you added completely discards that feature.